cypress-io/cypress
# fix: Fix issues with Cypress.require() and TypeScript

- Closes #25885

### Additional details

The root of the fix is two-fold:

- Moving the cy.origin() callback preprocessing into the child process, so it's the same as preprocessing the spec file itself.
- Using the user-installed version of the `typescript` package instead of our own, which actually doesn't exist in the production build of Cypress. This is also the same way it works when preprocessing the spec file itself.

### Steps to test

Set up a test project with a TypeScript spec file and use `Cypress.require()` within `cy.origin()`:

```typescript
describe('Some test', (): void => {
  it('should work', (): void => {
     cy.origin('https://example.com', (): void => {
        const test = Cypress.require('./test.js');
        cy.log('test', test);
     });
  });
});
```

### How has the user experience changed?


### PR Tasks


- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
